pkg/kubelet/kuberuntime/util/util.go

```diff
@@ -107,10 +109,13 @@ type RuntimeHandlerResolver interface {
 func NamespacesForPod(pod *v1.Pod, runtimeHelper kubecontainer.RuntimeHelper, rcManager RuntimeHandlerResolver) (*runtimeapi.NamespaceOption, error) {
 	runtimeHandler := ""
 	if pod != nil && rcManager != nil {
-		var err error
-		runtimeHandler, err = rcManager.LookupRuntimeHandler(pod.Spec.RuntimeClassName)
-		if err != nil {
-			return nil, err
+		val := reflect.ValueOf(rcManager)
+		if val.Kind() != reflect.Ptr || val.Kind() == reflect.Ptr && !val.IsNil() {
```

Comment:
I'm confused, can't this be implemented with a type assertion?

https://go.dev/play/p/n_LhBjDKTIh

```go
		rc, ok := rcManager.(runtimeclass.Manager)
		if ok {
		        // LookupRuntimeHandler...
		}
```

Reply:
Nvm, I see the issue, my solution will not work if the value is assigned an untyped pointer.

I wonder if we can have a function like this in `k8s.io/utils`:

```go
// isNilPtr checks if a value is a nil pointer. This is specially useful when dealing
// with interfaces that may hold pointers.
func isNilPtr(c any}) bool {
	return c == nil || (reflect.ValueOf(c).Kind() == reflect.Ptr && reflect.ValueOf(c).IsNil())
}
```

Reply:
> Nvm, I see the issue, my solution will not work if the value is assigned an untyped pointer.
> 
> I wonder if we can have a function like this in `k8s.io/utils`:
> 
> ```go
> // isNilPtr checks if a value is a nil pointer. This is specially useful when dealing
> // with interfaces that may hold pointers.
> func isNilPtr(p any) bool {
>         rv := reflect.Value(p)
> 	return p == nil || (rv.Kind() == reflect.Ptr && rv.IsNil())
> }
> ```

Sorry for the late reply, I agree with you. I will submit a PR to the `k8s.io/utils`. If it gets merged, I will then update this PR accordingly